src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -123,20 +132,51 @@ static String generateAppInstallationToken(String appId, String appPrivateKey, S
                 appInstallation = appInstallations.get(0);
             } else {
                 appInstallation = appInstallations.stream()
-                        .filter(installation -> installation.getAccount().getLogin().equals(owner))
-                        .findAny()
-                        .orElseThrow(() -> new IllegalArgumentException(String.format(ERROR_NOT_INSTALLED, appId)));
+                    .filter(installation -> installation.getAccount().getLogin().equals(owner))
+                    .findAny()
+                    .orElseThrow(() -> new IllegalArgumentException(String.format(
+                        ERROR_NOT_INSTALLED,
+                        appId)));
             }
 
             GHAppInstallationToken appInstallationToken = appInstallation
-                    .createToken(appInstallation.getPermissions())
-                    .create();
-
-            return appInstallationToken.getToken();
-        } catch (IOException e) {
+                .createToken(appInstallation.getPermissions())
+                .create();
+
+            long expiration = getExpirationSeconds(appInstallationToken);
+            LOGGER.log(Level.FINEST,
+                "Token raw expiration epoch seconds: {0,number,#}",
+                expiration);
```

Comment:
Best to avoid complex formats.

```suggestion
            LOGGER.finest(() -> "Token raw expiration epoch seconds: " + expiration);
```